Content.Server/Supermatter/EntitySystems/SupermatterSystem.cs

```diff
@@ -0,0 +1,425 @@
+            {
+                SMcomponent.AtmosUpdateAccumulator -= SupermatterComponent.AtmosUpdateTimer;
+                SMcomponent.Mix = mixture;
+                var pressure = mixture.Pressure;
+                var moles = mixture.TotalMoles;
+                var temp = mixture.Temperature;
+
+                if (moles > 0f)
+                {
+                    var Oxy = mixture.Moles[0];
+                    var Nit = mixture.Moles[1];
+                    var Co2 = mixture.Moles[2];
+                    var Pla = mixture.Moles[3];
+                    var Tri = mixture.Moles[4];
+                    var H2o = mixture.Moles[5];
+
+                    for(int gasId = 0; gasId < SMcomponent.GasComp.Length; gasId++)
+                    {
+                        SMcomponent.GasComp[gasId] = Math.Clamp(mixture.Moles[gasId] / moles, 0, 1);
+                    };
+
+                    var h2oBonus = 1 - (SMcomponent.GasComp[5] * 0.25f);
+
+
+                    SMcomponent.GasmixPowerRatio = 0f;
+                    SMcomponent.DynamicHeatModifier = 0f;
+                    SMcomponent.PowerTransmissionBonus = 0f;
+                    for(int gasId = 0; gasId < SMcomponent.GasComp.Length; gasId++)
+                    {
+                        //TODO: activate this once NO2/Hydrogen/proto nitrate is in
+                        //Value between 1 and 10. Effects the damage heat does to the crystal
+                        //SMcomponent.DynamicHeatResistance += SMcomponent.GasComp[gasId] + SMcomponent.gasFacts[gasId, 3];
+
+                        //No less then zero, and no greater then one, we use this to do explosions and heat to power transfer
+                        //Be very careful with modifing this var by large amounts, and for the love of god do not push it past 1
+                        SMcomponent.GasmixPowerRatio += SMcomponent.GasComp[gasId] * SMcomponent.gasFacts[gasId, 2];
+
+                        //Minimum value of -10, maximum value of 23. Effects plasma and o2 output and the output heat
+                        SMcomponent.DynamicHeatModifier += SMcomponent.GasComp[gasId] + SMcomponent.gasFacts[gasId, 1];
+
+                        //Value between -5 and 30, used to determine radiation output as it concerns things like collectors.
+                        SMcomponent.PowerTransmissionBonus += SMcomponent.GasComp[gasId] + SMcomponent.gasFacts[gasId, 0];
+                    };
+                    //SMcomponent.DynamicHeatResistance = Math.Max(SMcomponent.DynamicHeatResistance, 1);
+                    SMcomponent.DynamicHeatResistance = 1f;
+                    SMcomponent.GasmixPowerRatio = Math.Clamp(SMcomponent.GasmixPowerRatio, 0, 1);
+                    SMcomponent.DynamicHeatModifier = Math.Max(SMcomponent.DynamicHeatModifier, 0.5f);
+                    SMcomponent.PowerTransmissionBonus *= h2oBonus;
```

Comment:
making the SMcomponent variable name a little shorter would make this a lot more readable